BricksAndHearts.UnitTests/ServiceTests/Property/PropertyServiceTests.cs

```diff
@@ -1,31 +1,255 @@
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+        var propertiesBeforeCount = context.Properties.Count();
+        var deleteModel = context.Properties.Where(u => u.Id == 1).ToList()[0];
+
+        // Act
+        service.DeleteProperty(deleteModel);
+        context.ChangeTracker.Clear();
+
+        // Assert
+        context.Properties.Count(u => u.Id == 1).Should().Be(0);
+        context.Properties.Count().Should().Be(propertiesBeforeCount - 1);
+    }
+
+    #endregion
+
+    #region CountProperties()
+
+    [Fact]
+    public async void CountProperties_ReturnsCorrectCountsOfAll()
+    {
+        // Arrange
+        await using var context = Fixture.CreateReadContext();
+        var service = new PropertyService(context);
+
+        // Act
+        var propertyCount = service.CountProperties();
+        context.ChangeTracker.Clear();
+
+        // Assert
+        propertyCount.RegisteredProperties.Should().Be(context.Properties.Count());
+    }
+
+    #endregion
+
+    #region IsUserAdminOrCorrectLandlord
+
+    [Theory]
+    [InlineData(1)]
+    [InlineData(2)]
+    public async void IsUserAdminOrCorrectLandlord_UsedByAdmin_ReturnsTrueEvenIfPropertyIsntTheirs(int OwnerId)
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+
+
+        var adminUser = CreateAdminUser();
+
+
+        context.Properties.First(u => u.Id == 1).LandlordId = OwnerId;
+
+        // Act
+        var allowed = service.IsUserAdminOrCorrectLandlord(adminUser, 1);
+        context.ChangeTracker.Clear();
+
+        // Assert
+        allowed.Should().BeTrue();
+    }
+
+
     [Fact]
-    public async void CountProperties_ReturnsPropertyCountModel_WithCorrectData()
+    public async void IsUserAdminOrCorrectLandlord_UsedByLandLord_ReturnsTrueIfPropertyIsTheirs()
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+        var landlordUser = CreateLandlordUser();
+        landlordUser.LandlordId = 50;
+
+        context.Properties.First(u => u.Id == 1).LandlordId = 50;
+
+        // Act
+        var allowed = service.IsUserAdminOrCorrectLandlord(landlordUser, 1);
+        context.ChangeTracker.Clear();
+
+        // Assert
+        allowed.Should().BeTrue();
+    }
+
+    [Fact]
+    public async void IsUserAdminOrCorrectLandlord_UsedByLandLord_ReturnsFalseIfPropertyIsNotTheirs()
+    {
+        // Arrange
+        await using var context = Fixture.CreateWriteContext();
+        var service = new PropertyService(context);
+        var landlordUser = CreateLandlordUser();
+        landlordUser.LandlordId = 50;
+
+        context.Properties.First(u => u.Id == 1).LandlordId = 49;
+
+        // Act
+        var allowed = service.IsUserAdminOrCorrectLandlord(landlordUser, 1);
+        context.ChangeTracker.Clear();

```

Comment:
Don't need the change tracker clear here. If you wanted to be extra careful you could save the property change and clear the change tracker before the `Act` line

Reply:
I think I misread the part of Max's guide about changetrackers 😅